keela-widgets/cameramanager.cpp

```diff
@@ -342,3 +320,23 @@ void Keela::CameraManager::remove_frame_splitting_probes() {
 		remove_probe_by_id(odd_frame_probe_id, odd_sink_pad, "odd frame filter");
 	}
 }
+
+void Keela::CameraManager::set_pipeline_state(GstState state) {
```

Comment:
this isn't really quite doing what it says on the tin (this sets the camera element's state, not the state of the entire pipeline, but this stalls the pipeline anyway) but it does produce the desired effect. Perhaps this belongs as a class method on Keela::Element so that other GstElements have this capability as well

Reply:
That makes sense, your downstream caps do depend on the upstream camera, so we'd need to retrigger caps negotiation